src/presentation/api/routers/assignments.py

```
@@ -0,0 +1,213 @@
+        raise HTTPException(status_code=404, detail="Not found")
+    # Safely obtain payload dict (see create_assignment note)
+    if hasattr(data, "model_dump"):
+        payload = data.model_dump(exclude={"id"})
+    else:
+        payload = data.dict(exclude={"id"})
+    # Ensure weighted_mark is numeric (float) when provided
+    if payload.get("weighted_mark") is not None:
+        try:
+            payload["weighted_mark"] = float(payload["weighted_mark"])
+        except Exception:
+            pass
```

Ruff (0.14.3)

`try`-`except`-`pass` detected, consider logging the exception

(S110)